### PR TITLE
Detail endpoint

### LINKED ISSUE #130: API: Endpoint to retrieve the detail for a specific event
Create an API endpoint that returns the detail for a specific event so that the frontend can show it - check #129 

image

The endpoint should receive:

- event id

and it should then filter for that specific event and return:

- Organization Name, 
- Week number, 
- Date for the events, 
- Registered Mentor, 
- Link for the event, 
- Names of the registered students, 
- Topics of the students, 
- Link of the code of the students

The endpoint has to be documented on Swagger

### PR DESCRIPTION
**Pull request by nuhana**
# Description

Endpoint to retrieve the detail for a specific event

Fixes #130 

# How to test?

Swagger and Postman

# Checklist

- [x] I have performed a self-review of my own code

### CODE REVIEW

File: src/server/api/routes/api-router.js [hunk 1]
```diff
@@ -52,6 +53,11 @@ router.use('/modules', modulesRouter);
 router.use('/organizations', organizationsRouter);
 router.use('/profile', profileRouter);
 router.use('/self', selfRouter);
+
+router.use('/events_users', eventsUsersRouter2);
```
Comment (tpetrina) [inline src/server/api/routes/api-router.js]: Remove this line.

File: src/server/api/routes/api-router.js [hunk 2]
```diff
@@ -52,6 +53,11 @@ router.use('/modules', modulesRouter);
...
+
+router.use('/eventDetail', eventsDetailRouter);
```
Comment (tpetrina) [inline src/server/api/routes/api-router.js]: Why not put it under `/events/:eventId/details`?

File: src/server/api/controllers/eventDetail.controller.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+const eventDetail = async (eventId) => {
+  const details = await knex('events')
+    .select(
+      'users.name',
+      'questions.topic',
+      'questions.description',
+      'events.event_date',
+      'topics.topic_name',
+      'topics.week_number',
+      'topics.homework_url',
```
Comment (tpetrina) [inline src/server/api/controllers/eventDetail.controller.js]: The output contains `_` and you should rename it to `camelCase`.

**Review by tpetrina — CHANGES_REQUESTED** [review]
Good job, small changes needed.

File: src/server/api/routes/api-router.js (hunk 2 above)
Comment (nuhana) [reply src/server/api/routes/api-router.js]: @tpetrina  i got 'Sorry can't find that!' when tried to use this

Comment (tpetrina) [reply src/server/api/routes/api-router.js]: @nuha90 Well, it depends on what exactly did you change and how the full route is constructed.

File: src/server/api/controllers/eventDetail.controller.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+const eventDetail = async (eventId) => {
+  const details = await knex('events')
+    .select(
+      'users.name as Student name',
+      'questions.description',
+      'events.event_date as Event date',
+      'topics.topic_name as Topic',
+      'topics.week_number as Week number',
+      'topics.homework_url as Link',
+      'organizations.name as Organization name',
```
Comment (tpetrina) [inline src/server/api/controllers/eventDetail.controller.js]:
```suggestion
      'users.name as name',
      'questions.description',
      'events.event_date as eventDate',
      'topics.topic_name as topic',
      'topics.week_number as weekNumber',
      'topics.homework_url as link',
      'organizations.name as organization',
```
This is `camelCase`. JSON properties usually do not have spaces in them.

**Review by tpetrina — CHANGES_REQUESTED** [review]

File: src/server/api/controllers/eventDetail.controller.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+const eventDetail = async (eventId) => {
+  const details = await knex('events')
+    .select(
+      'users.name as name',
+      'questions.description',
+      'events.event_date as eventDate',
+      'topics.topic_name as topic',
+      'topics.week_number as weekNumber',
+      'topics.homework_url as link',
+      'organizations.name as Organization',
```
Comment (tpetrina) [inline src/server/api/controllers/eventDetail.controller.js]:
```suggestion
      'organizations.name as organization',
```

Lowercase `o`.

File: src/server/api/routes/api-router.js (hunk 1 above)
Comment (tpetrina) [reply src/server/api/routes/api-router.js]: Please remove this line @nuha90

**Review by tpetrina — CHANGES_REQUESTED** [review]
One more removal.

**Review by tpetrina — APPROVED** [review]
Good job!

**Review by pallabi-ghosh — APPROVED** [review]

**Pull request merged**